saleor/graphql/checkout/mutations.py

```diff
@@ -733,19 +744,23 @@ def perform_mutation(cls, _root, info, checkout_id, store_source, **data):
                     {"redirect_url": error}, code=AccountErrorCode.INVALID
                 )
 
-        # create the order into the database
-        order = create_order(
-            checkout=checkout,
-            order_data=order_data,
-            user=user,
-            redirect_url=redirect_url,
-        )
+        order = None
+        if not txn.action_required:
+            # create the order into the database
+            order = create_order(
+                checkout=checkout,
+                order_data=order_data,
+                user=user,
+                redirect_url=redirect_url,
+            )
+
+            # remove checkout after order is successfully paid
+            checkout.delete()
 
-        # remove checkout after order is successfully paid
-        checkout.delete()
+            # return the success response with the newly created order data
+            return CheckoutComplete(order=order)
 
-        # return the success response with the newly created order data
-        return CheckoutComplete(order=order)
+        return CheckoutComplete(order=None, confirmation_needed=True)
```

Comment:
Don't we need a separate mutation for payment that needs confirmation?
Now we have `CheckoutComplete` but we don't complete the checkout as we need to do additional steps.

Reply:
What should be the next step in the checkout flow when we get `CheckoutComplete(order=None, confirmation_needed=True)`?

Reply:
You confirm payment client side (depends on gateway) and call again.

Reply:
So to finish checkout I need to execute mutation `CheckoutComplete` twice?

Reply:
yes, in case there is additional confirmation required, you need to call mutation again